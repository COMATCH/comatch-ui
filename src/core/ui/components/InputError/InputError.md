
## InputError

From [`InputError.js`](InputError.js)

An error indicator next to inputs for client side validation errors on form submissions.

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**text** | `String` |  | :white_check_mark: | Text content of `InputError`



