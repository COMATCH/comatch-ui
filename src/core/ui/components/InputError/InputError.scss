@import '../../../styles/variables.scss';

// Explanatory error message
// for erroneous Input components.
.InputError {
    color: $co-color-red;
    display: block;
    font-size: $co-font-sm;
    margin: 7px 0 10px 0;
}
