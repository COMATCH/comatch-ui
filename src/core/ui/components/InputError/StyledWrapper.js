import styled from 'styled-components';
import { palette, typography } from '../../../styles/variables';

const StyledWrapper = styled.span`
    color: ${palette.red};
    display: block;
    font-size: ${typography.fontSm};
    margin: 7px 0 10px 0;
`;

export { StyledWrapper };
