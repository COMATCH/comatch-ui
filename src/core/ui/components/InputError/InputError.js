import React from 'react';
import PropTypes from 'prop-types';

import { StyledWrapper } from './StyledWrapper';

const propTypes = {
    /**
     * Text content of `InputError`
     */
    text: PropTypes.string.isRequired,
};

const defaultProps = {};

/**
 * An error indicator next to inputs for client side validation errors on form submissions.
 */

export const InputError = ({ text }) => (
    <StyledWrapper className="InputError">{text}</StyledWrapper>
);

InputError.defaultProps = defaultProps;

InputError.propTypes = propTypes;
