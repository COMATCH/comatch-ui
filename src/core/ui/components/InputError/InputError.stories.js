import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { InputError } from './InputError';

storiesOf('InputError', module).add('with basic style', () => (
    <>
        <InputError text="Sample error" />
        <Highlight className="html">{'<InputError text="Sample error" />'}</Highlight>
    </>
));
