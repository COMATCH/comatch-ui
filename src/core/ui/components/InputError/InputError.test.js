import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { InputError } from './InputError';
import '../../../config/tests/setup';

const testText = 'Testing InputError';

describe('InputError', () => {
    it('should render correctly', () => {
        const inputError = renderer.create(<InputError text={testText} />).toJSON();
        expect(inputError).toMatchSnapshot();
    });

    it('should render an InputError with default props', () => {
        const item = shallow(<InputError text={testText} />);

        // should have a default classe .InputError
        expect(item.hasClass('InputError')).toEqual(true);
    });

    it('should render an InputError with specified text', () => {
        const item = shallow(<InputError text={testText} />);

        expect(item.text()).toEqual(testText);
    });
});
