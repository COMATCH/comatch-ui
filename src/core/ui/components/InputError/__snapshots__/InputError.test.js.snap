// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError should render correctly 1`] = `
<span
  className="InputError"
>
  Testing InputError
</span>
`;
